Stripe/STPSMSCodeViewController.m

```diff
@@ -108,10 +109,38 @@ - (void)viewDidLoad {
     [self.scrollView addSubview:smsSentLabel];
     self.smsSentLabel = smsSentLabel;
     
+    UIButton *pasteFromClipboardButton = [UIButton buttonWithType:UIButtonTypeSystem];
+    pasteFromClipboardButton.titleLabel.textAlignment = NSTextAlignmentCenter;
+    [pasteFromClipboardButton setTitle:NSLocalizedString(@"Paste copied code?", nil) forState:UIControlStateNormal];
+    [pasteFromClipboardButton addTarget:self action:@selector(pasteCodeFromClipboard) forControlEvents:UIControlEventTouchUpInside];
+    pasteFromClipboardButton.alpha = 0;
+    pasteFromClipboardButton.hidden = YES;
+    [self.scrollView addSubview:pasteFromClipboardButton];
+    self.pasteFromClipboardButton = pasteFromClipboardButton;
+    
     STPPaymentActivityIndicatorView *activityIndicator = [STPPaymentActivityIndicatorView new];
     [self.scrollView addSubview:activityIndicator];
     _activityIndicator = activityIndicator;
     [self updateAppearance];
+    
+    NSNotificationCenter *notificationCenter = [NSNotificationCenter defaultCenter];
+    [notificationCenter addObserver:self 
+                           selector:@selector(applicationDidBecomeActive) 
+                               name:UIApplicationDidBecomeActiveNotification 
+                             object:nil];
+}
+
+- (void)applicationDidBecomeActive {
+    if (self.view.superview != nil) {
```

Comment:
It checks to make sure the view is still actually on screen and visible. It's mostly in there because I copied this from another screen in the app that did the same check, but maybe not necessary.